lib/new_relic/agent/instrumentation/grape.rb

```diff
@@ -19,9 +19,13 @@
 
   depends_on do
     begin
-      if defined?(Bundler) &&
-          ((Bundler.rubygems.respond_to?(:installed_specs) && Bundler.rubygems.installed_specs.map(&:name).include?('newrelic-grape')) ||
-            Bundler.rubygems.all_specs.map(&:name).include?('newrelic-grape'))
+      specs = if Bundler.rubygems.respond_to?(:installed_specs)
+        Bundler.rubygems.installed_specs
+      else
+        Bundler.rubygems.all_specs
+      end
+
+      if specs.map(&:name).include?('newrelic-grape')
```

Comment:
Maybe a safe navigation operator before map JIC specs is nil? 

I took out the defined?(Bundler) check because if it raises an error, the error should be rescued within this block, but happy to add it back in if that's preferable.

Reply:
I think this is good as is, given that all known Bundler versions will return an array that can handle `#map`, empty or not. If you really want to try to harden and futureproof the logic, we could:

- Use `if/elsif/else` to check for `#all_specs` to be defined with an `elsif` and `else` into an early `false` return if neither are defined.
- Move the logic out into a helper shared by all similar Bundler specs checks

But both of those ideas seem like overkill and out of place stylistically for the norm for dependency detection checks which like to be simple and self-contained.

Reply:
Since this isn't tested, I like the idea of moving this out into a helper shared by all similar Bundler specs checks. The lack of tests here feels wrong, but the mechanics with multiverse or mocks seemed heavy/unreliable.

We have similar spec gathering logic in four separate locations. Getting the logic to create the specs array in one place may make it easier to test. I'll give that a try tomorrow. Thank you!

Reply:
Updated in:  9cdcd2c33c312acbcf72161ca9dbafe728b495a9